Services/Certificate/classes/Placeholder/Values/class.ilCoursePlaceholderValues.php

```diff
@@ -1,70 +1,29 @@
-<?php
+<?php declare(strict_types=1);
 /* Copyright (c) 1998-2018 ILIAS open source, Extended GPL, see docs/LICENSE */
 
 /**
  * @author  Niels Theen <ntheen@databay.de>
  */
 class ilCoursePlaceholderValues implements ilCertificatePlaceholderValues
 {
-    /**
-     * @var ilDefaultPlaceholderValues
-     */
-    private $defaultPlaceHolderValuesObject;
-
-    /**
-     * @var ilObjectCustomUserFieldsPlaceholderValues
-     */
-    private $customUserFieldsPlaceholderValuesObject;
-
-    /**
-     * @var ilLanguage|null
-     */
-    private $language;
+    private ?ilDefaultPlaceholderValues $defaultPlaceHolderValuesObject;
+    private ?ilObjectCustomUserFieldsPlaceholderValues $customUserFieldsPlaceholderValuesObject;
+    private ?ilLanguage $language;
+    private ?ilCertificateObjectHelper $objectHelper;
+    private ?ilCertificateParticipantsHelper $participantsHelper;
+    private ?ilCertificateUtilHelper $ilUtilHelper;
+    private ?ilCertificateDateHelper $dateHelper;
+    private ?ilCertificateLPStatusHelper $lpStatusHelper;
```

Comment:
These types are not optional. Only the passed constructor arguments are optional.